janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -1764,6 +1764,67 @@ public void testIndexUpdateSyncWithMultipleInstances() throws InterruptedExcepti
 
     }
 
+    @Category({BrittleTests.class})
+    @Test
+    public void testIndexSHouldRegisterWhenWeRemoveAnInstance() throws InterruptedException {
+        clopen(option(LOG_SEND_DELAY, MANAGEMENT_LOG), Duration.ofMillis(0),
+                option(KCVSLog.LOG_READ_LAG_TIME, MANAGEMENT_LOG), Duration.ofMillis(50),
+                option(LOG_READ_INTERVAL, MANAGEMENT_LOG), Duration.ofMillis(250)
+        );
+
+        StandardJanusGraph graph2 = (StandardJanusGraph) JanusGraphFactory.open(config);
+        JanusGraphTransaction tx2;
+
+        mgmt.makePropertyKey("name").dataType(String.class).make();
+        finishSchema();
+
+        tx.addVertex("name", "v1");
+        newTx();
+        evaluateQuery(tx.query().has("name", "v1"), ElementCategory.VERTEX, 1, new boolean[]{false, true});
+        tx2 = graph2.newTransaction();
+        evaluateQuery(tx2.query().has("name", "v1"), ElementCategory.VERTEX, 1, new boolean[]{false, true});
+        //Leave tx2 open to delay acknowledgement
+
+        mgmt.buildIndex("theIndex", Vertex.class).addKey(mgmt.getPropertyKey("name")).buildCompositeIndex();
+        mgmt.commit();
+
+        JanusGraphTransaction tx3 = graph2.newTransaction();
+        tx3.addVertex("name", "v2");
+        tx3.commit();
+        newTx();
+        tx.addVertex("name", "v3");
+        tx.commit();
+
+        Thread.sleep(2000); //Wait for the index to register in graph2
+        finishSchema();
+        try {
+            mgmt.updateIndex(mgmt.getGraphIndex("theIndex"), SchemaAction.ENABLE_INDEX);
+            fail(); //Open tx2 should not make this possible
+        } catch (IllegalArgumentException e) {
```

Comment:
I'm trying to follow this... the attempt to move `theIndex` to ENABLED throws an exception because the transaction is still open. That makes sense, but is `theIndex` in REGISTERED status at this point, or is it INSTALLED status? The comment below on line 1807 (close second graph instance, so index can move to REGISTERED) is adding to my confusion since I would think that it is already REGISTERED based on the comment on line 1798 (Wait for the index to register in graph2). https://github.com/JanusGraph/janusgraph/wiki/Indexing

Reply:
I think this test was failing for one of the backends; if I remember correctly, the backend returning the `graph.getOpenInstances()` wasn't returning "graph2" yet, and we needed to allow it time to catch up (otherwise the index will indeed move to REGISTERED and the index will not throw an exception when we try to move it to ENABLED). So this comment is _not_ a testament to the index status rather the state of the system's open instances. 

I am going to try removing the line for now and see what happens.

Reply:
removed the line and tests pass so all is good in the world